### PR TITLE
Remove warnings, errors and improve Jest specs

### PR DESCRIPTION
**Pull request by josemigallas**
* Remove errors and warnings that were showing up in the log
* Improve some tests
* Remove some _todo_ and _skip_

### CODE REVIEW

File: spec/javascripts/Stats/inlinechart/inlineChart.spec.jsx [hunk 1]
```diff
@@ -89,26 +91,13 @@ describe('<InlineChart/>', () => {
   })
 
   it('calls updateState method with total of 1', () => {
-    const data = {
```
Comment (damianpm) [inline spec/javascripts/Stats/inlinechart/inlineChart.spec.jsx]: I think this tests is useless, after removing `data`, you are calling it with the default data object where `total: 10`, then it's just asserting `toHaveBeenCalled` but not with `data:1`

File: spec/javascripts/Stats/inlinechart/inlineChart.spec.jsx [hunk 2]
```diff
@@ -89,26 +91,13 @@ describe('<InlineChart/>', () => {
...
-    }
-
     wrapper.instance().generateC3Chart = jest.fn()
     wrapper.update()
     wrapper.instance().updateState(data)
     expect(wrapper.instance().generateC3Chart).toHaveBeenCalled()
   })
 
   it('calls updateState method with total > 1', () => {
-    const data = {
```
Comment (damianpm) [inline spec/javascripts/Stats/inlinechart/inlineChart.spec.jsx]: Same here, it's not asserting `total > 1`

File: spec/javascripts/Stats/inlinechart/inlineChart.spec.jsx (hunk 1 above)
Comment (josemigallas) [reply spec/javascripts/Stats/inlinechart/inlineChart.spec.jsx]: You're right, I didn't realize the "total" was different.. the test is wrong then, it should fail. I will fix it.

File: app/javascript/src/Policies/components/PoliciesForm.jsx [hunk 3]
```diff
@@ -10,6 +10,7 @@ import type { ChainPolicy } from 'Policies/types/Policies'
 import type { UpdatePolicyConfigAction } from 'Policies/actions/PolicyConfig'
 
 class PolicyForm extends Form {
+  // TODO: this method will be renamed to UNSAFE_componentWillReceiveProps in 1.9 and eventually removed.
```
[the remark this answers is not in the document]
Comment (josemigallas) [reply app/javascript/src/Policies/components/PoliciesForm.jsx]: Done

**Pull request merged**